### DISCUSSION TITLE
pvoutput.org feature request

### DISCUSSION THREAD

**Discussion by TheLordFolken** [wlcrs/huawei_solar#113] [Ideas]
Is it possible to have this integration also upload data to pvoutput? pvoutput has a relatively simple API but I lack the programming ability to do this myself. API specifications can be found at https://www.pvoutput.org/help/api_specification.html

**Comment by wlcrs** [comment]
This is not within scope of this integration. Generic solutions that can be coupled to any HA entity (and thus any solar system that is integrated into HA) exist: https://community.home-assistant.io/t/pvouput-uploader/46761

**Reply by TheLordFolken** [reply]
Thank you very much, the suggestion solutions works well.

**Reply by TheLordFolken** [reply]
Hi Mate,
Can you help me with the logic? pvoutput has a value for 'Power Consumption (watts)'. Would I be correct that the value is the sum of:
battery discharge + grid import + (input power - grid export). It also has a value for 'Power Generation (watts)' which I'm assuming is input_power.

As an aside, do you lurk on discord or a chat platform?

Thanks,
Trav

**Reply by WildEchoWanderer** [reply]
> Hi Mate, Can you help me with the logic? pvoutput has a value for 'Power Consumption (watts)'. Would I be correct that the value is the sum of: battery discharge + grid import + (input power - grid export). It also has a value for 'Power Generation (watts)' which I'm assuming is input_power.
> 
> As an aside, do you lurk on discord or a chat platform?
> 
> Thanks, Trav

Did you get it to work? :)
I try to config my Energy dashboard and i am not understanding the infos in the link

**Comment by Niek** [comment]
See also: https://www.bazmac.me/blog/setting-up-home-assistant-to-upload-to-pvoutput

If someone figured out the right `v*` values (see https://pvoutput.org/help/api_specification.html#add-status-service) to set, please reply here.
